Mathlib/Tactic/Lift.lean

```diff
@@ -159,12 +159,24 @@ def Lift.main (e t : TSyntax `term) (hUsing : Option (TSyntax `term))
     evalTactic (← `(tactic| try clear $hUsingStx))
   if hUsing.isNone then withMainContext <| setGoals (prf.mvarId! :: (← getGoals))
 
+open Linter in
 elab_rules : tactic
 | `(tactic| lift $e to $t $[using $h]? $[with $newVarName $[$newEqName]? $[$newPrfName]?]?) =>
-  withMainContext <|
-    let keepUsing := match h, newPrfName.join with
-      | some h, some newPrfName => h.raw == newPrfName
-      | _, _ => false
+  withMainContext <| do
+    let keepUsing ← do
+      if let .some newPrfName := newPrfName.join then
+        if h.map (·.raw) == some newPrfName then
+          if getLinterValue linter.deprecated (← getLinterOptions) then
+            Lean.logWarningAt newPrfName <|
+              .tagged ``deprecatedAttr
+                m!"This syntax is deprecated; to avoid clearing {h}, \
+                  replace it with `id {h}` instead."
+          pure true
+        else
+          logErrorAt newPrfName "This syntax doesn't do anything, please remove it."
```

Comment:
```suggestion
          logWarningAt newPrfName "This syntax doesn't do anything, please remove it."
```
What do you think?  I personally find that logging errors should also imply that something should fail.  I.e., logging an error should imply a `throwError`/`failure` *somewhere*.

Regardless of this, maybe there should also be a test exercising the branch of the code.